### PR TITLE
nixos/rtorrent: add rtorrent tmux service (user and system)

### PR DESCRIPTION
**Pull request by csingley**
###### Motivation for this change
rtorrent systemd service units, started within a tmux session.  Both a system service and a user service are provided, somewhat following the install/enable config syntax of the emacsd expression.  Particularly handy to bring up rtorrent at boot time.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by adisbladis** [comment]
This PR should not contain commits from https://github.com/NixOS/nixpkgs/pull/30144 .
A lot of these commits should get squashed, I can't see a reason why this should be more than one commit.

File: nixos/modules/services/torrent/rtorrent.nix [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        If true, services.rtorrent.install is considered true, whatever its value.
+      '';
+    };
+    install = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to install a user service for rtorrent (via tmux).
+        The service must be manually started for each user with
```
Comment (adisbladis) [inline nixos/modules/services/torrent/rtorrent.nix]: Duplicate text.

File: nixos/modules/services/torrent/rtorrent.nix [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          cat > "${cfg.configFile}" << EOF
+      '' + replaceStrings [ "/home/USERNAME/rtorrent" ] [ cfg.dataDir ] (readFile ./rtorrent.rc) + "\nEOF\n}";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "forking";
+        KillMode = "none";
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = "${pkgs.tmux}/bin/tmux new-session -c ${cfg.dataDir} -s rtorrent -n rtorrent -d rtorrent -n -o import=${cfg.configFile}";
+        ExecStop = "${pkgs.bash}/bin/bash -c \"tmux send-keys -t rtorrent C-q && while pidof rtorrent > /dev/null; do sleep 0.5; done\"";
```
Comment (adisbladis) [inline nixos/modules/services/torrent/rtorrent.nix]: This looks very hacky. Not sure if there is a better way to solve this together with tmux but it doesn't feel right to me.

Comment (csingley) [reply nixos/modules/services/torrent/rtorrent.nix]: Yeah it's a little reminiscent of creaky init scripts.
On the plus side, it's at least a fairly well tested hack; lots of people have been using it on Arch for a few years:

https://wiki.archlinux.org/index.php/RTorrent#Systemd_services_using_tmux_or_screen

**Comment by csingley** [comment]
Looks like the rtorrent devs have merged a patch to run rtorrent as a headless daemon, accessible only via XML-RPC not CLI:

https://github.com/rakshasa/rtorrent/pull/446

This would be a better candidate, allowing systemd to manage the service directly and saving all this jiggery-pokery of operating through tmux.  I'll withdraw the PR and wait for the release of 0.9.7.

**Pull request closed without merging**